### ISSUE TITLE
nytimes data no longer updating

### ISSUE THREAD

**Issue by jgoerzen** [cipriancraciun/covid19-datasets#43]
Hi,

Thanks for this resource!  It looks, by the way, like the New York Times data is no longer updating.  Last update was 4 days ago.  Are you able to restore that data?

Thanks!

**Comment by cipriancraciun** [comment]
Indeed none of the datasets were updated since the weekend.  Until a month ago I was regenerating the datasets each day, but now given that it seems it would take "a while" until the COVID situation is resolved, and the fact that I'm nearing the GitHub maximum repository size, I thought I'll re-generate it only once or twice a week.

I think I'll schedule the regeneration on Saturday / Sunday to capture the previous week and on Wednesday / Thursday.

Would that work for you?

----

BTW, the execution now takes quite some time, nearly 4 hours, and it eats almost 32 GB of RAM, that it will become quite an issue in a few months, and at the moment I don't think I have the time to rewrite the JQ / Python scripts in a more efficient fashion...

**Comment by jgoerzen** [comment]
Totally understood.  I would really like to get this daily.  I have spent some time trying to figure out how to make that happen, and although I did get z-run and ninja installed, running the script brings up a menu of options, none of which work.  Would you perhaps be able to write up a little documentation on how to run things?  That would be fantastic for my needs.

Thanks again for all your work on this!

**Comment by cipriancraciun** [comment]
@jgoerzen In order to have it running you need a few non-standard tools and a few standard ones installed.  On the top of my head the following are the requirements:
* Linux -- this is almost mandatory;  it could perhaps work on OSX, but with a lot of effort due to the standard tools mismatch between OSX and Linux (i.e. `cp`, `mv` and other similar ones);  (if you need OSX we can discuss this further;)
* `z-run` -- https://github.com/cipriancraciun/z-run -- a personal tool, take the latest version (I've pushed something yesterday);  to build it you need to go in the `sources` folder and just `go build -o ./z-run ./cmd/z-run.go`;  (also update the `covid19-datasets` repository to have the latest `z-run` scripts that adapt for that change;)
* `python2.7`, together with a few libraries installed (I think only `unidecode`;)  I recommend installing it from the default repositories (that is `python2` and `python2-Unidecode` on OpenSUSE), else if you use a virtual environment make sure that that `python2` executable is exposed into the `PATH` of the environment;
* `jqe` -- a patched version of `jq` -- https://github.com/cipriancraciun/jq/tree/patches/latest -- take the `patches/latest` branch;  to build it you need `liblmdb-dev` (or equivalent), and the OpenSSL dev packages;  (I doubt it will build on OSX;)
* `ninja` -- this is the standard one;
* `nodaemon` -- if you get an error that this is missing just `ln -s /usr/bin/env /usr/local/bin/nodaemon`;  (it should be required only for `julia` which is required for graphs;)
* try to run it, and if it fails, keep installing missing tools;  (except the `z-run` and `jqe` above, all the rest are "plain" ones;)

Also, given that I run the build remotely, it tries by default to SSH into a build machine.  To disable that just `export __ninja_processor=local` in the local environment.

To build the datasets you should look inside the `ninja / targets / ...` menu.  You can also run without the menu for example `z-run ':: ninja / targets / nytimes / exports`' to build the NyTimes datasets.

Also daily you have to manually edit the `scripts/datasets-jhu.z-run` file to manually add the latest dates available;  I didn't automate this because the JHU didn't have a stable release schedule.

Please note it takes a lot of time and a lot of RAM.  (The main culprit is the `jq` tool which was not designed for such amount of data, but rewriting that means scratching almost everything and starting from zero.)

If you encounter any issues running it, please open another ticket and paste the errors you get.

(BTW, today being Monday I'm rebuilding the datasets and will push them in a few hours.)

**Comment by cipriancraciun** [comment]
@jgoerzen Have you managed to run the scripts independently?

(I'm now regenerating the datasets;  it will take a couple of hours to finalize.)

**Comment by jgoerzen** [comment]
Unfortunately I've been extremely short on time and haven't had the chance to work with it more.  I apologize.

What I need is just the SQLite extracts (and it doesn't have to be the combined version).  I also need it to be completely automated (non-interactive).  Unfortunately if it takes a number of hours and 32GB RAM then I probably can't make it work with Github Actions, and may have to rewrite the entire thing.  My goal is to have it run at least daily, automatically, in Github.

**Comment by cipriancraciun** [comment]
Which dataset do you need?  There are the following:
* NY Times, US only, states level;  (takes ~1 minute to run;)
  * `z-run 'ninja / files / build / exports/nytimes/v1/us-states/values-sqlite.sql.gz'`
* NY Times, US only, counties level;  (takes +2-3h to run;)
  * `z-run 'ninja / files / build / exports/nytimes/v1/us-counties/values-sqlite.sql.gz'`
* JHU, world country level, plus US states;  (takes ~4 minutes to run;)
  * `z-run ':: ninja / files / build / exports/jhu/v1/series/values-sqlite.db.gz'`
* JHU, world country level, plus US counties;  (takes +2-3h to run;)
  * `z-run ':: ninja / files / build / exports/jhu/v1/daily/values-sqlite.sql.gz'`
* ECDC, world country level;  (takes ~3 minutes;)
  * `z-run ':: ninja / files / build / exports/ecdc/v1/worldwide/values-sqlite.sql.gz'`

If you try to automate the building, I would suggest removing the folders `exports` and `imports`, but keeping the `imports/facts` folder: `find ./imports ./exports -mindepth 1 -not -path './imports/facts' -not -path './imports/facts/*' -delete`

If you are OK without the large datasets (that is NY Times and JHU with US county level), I could start regenerating them daily.

**Comment by cipriancraciun** [comment]
I've prepared some shortcuts that create only the "small" datasets, and given they execute quite quickly, I'll restart doing these daily.  (That is without NY Times and JHU with US country level.)
